ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/payloadattestation/MatchingDataPayloadAttestationGroup.java

```diff
@@ -37,8 +37,8 @@
  */
 class MatchingDataPayloadAttestationGroup {
 
-  private final Set<PayloadAttestationMessage> payloadAttestationMessages =
-      ConcurrentHashMap.newKeySet();
+  private final Int2ObjectMap<PayloadAttestationMessage> payloadAttestationMessages =
```

Comment:
We could remove synchronization by changhing this to

```
private final ConcurrentMap<Integer, PayloadAttestationMessage> payloadAttestationMessages =
      new ConcurrentHashMap<>();
```

and use `putIfAbsent` in the `add` method.

I'm lately a fan of removing the synchronization with a tiny boxing cost. WDYT?

Reply:
Oh yes, I like it. I was overthinking it with using `Int2ObjectMap` but can actually use that when creating the aggregate. Pushed a change.